app/grandchallenge/components/models.py

```diff
@@ -2046,3 +2046,43 @@ def create_civ_for_file(self, *, ci, current_civ, new_value):
             # if no new value is provided (user selects '---' in dropdown)
             # delete old CIV
             self.values.remove(current_civ)
+
+
+class ValuesForInterfacesMixin:
+    @cached_property
+    def values_for_interfaces(self):
+        from grandchallenge.archives.models import Archive
+        from grandchallenge.reader_studies.models import ReaderStudy
+
+        if self.__class__ == ReaderStudy:
+            qs = self.display_sets.select_related(
+                "values", "values__interface", "values__image"
+            )
+        elif self.__class__ == Archive:
+            qs = self.items.select_related(
+                "values", "values__interface", "values__image"
+            )
```

Comment:
This is a bit backwards as the Mixin needs to know about all its consumers. The way to do this would be something like:

```python
class Mixin:
  @property
  def related_attr(self): # think of a better name
    raise NotImplementedError
  
  def values_for_interfaces(self):
    queryset = self.related_attr.select_related(
                 "values", "values__interface", "values__image"
             )

...

class Consumer(Mixin):
  @property
  def related_attr(self):
    return self.items # or display_sets
```

Reply:
Ok, makes sense. And further below instead of  checking `self.__class__ == ReaderStudy` when adding the question interfaces for reader studies, a try ... except statement would be better?
``

Reply:
See the other suggestion, I think it's better to be explicit, e.g.

```python

class Mixin:
  @property
  def relevant_interfaces(self): # Again, naming...
    return self.related_attr.select_related(
                 "values", "values__interface", "values__image"
             )

  def values_for_interfaces(self):
    interfaces = self.relevant_interfaces
    ... # Do stuff with the interfaces

class Archive(Mixin):
  pass # we're happy with the default

class ReaderStudy(Mixin):
  @property
  def relevant_interfaces(self):
    interfaces = super().relevant_interfaces
    interfaces += self.questions.filter(...)
    return interfaces
```

Reply:
Never mind, got my answer in your other comment